sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Scenarios/DownloadFile.cs

```diff
@@ -0,0 +1,151 @@
+﻿//Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Test.Perf;
+
+namespace Azure.Storage.Files.Shares.Perf.Scenarios
+{
+    /// <summary>
+    /// The performance test scenario focused on downloading files from the Files Shares storage.
+    /// </summary>
+    /// <seealso cref="Azure.Test.Perf.PerfTest{SizeOptions}" />
+    public sealed class DownloadFile : PerfTest<SizeOptions>
+    {
+        /// <summary>
+        /// Name of the directory client.
+        /// </summary>
+        private string DirectoryClientName { get; set; }
+
+        /// <summary>
+        /// Name of the file client.
+        /// </summary>
+        private string FileClientName { get; set; }
+
+        /// <summary>
+        /// The client for interaction with the Files Shares file system.
+        /// </summary>
+        private ShareClient FilesShareClient { get; set; }
+
+        /// <summary>
+        /// The path of the file that is created locally from the downloaded data.
+        /// </summary>
+        private string DownloadFilePath { get; set; }
+
+        /// <summary>
+        /// File uploader.
+        /// </summary>
+        private readonly UploadFile _fileUploader;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DownloadFile"/> class.
+        /// </summary>
+        /// <param name="options">The set of options to consider for configuring the scenario.</param>
+        public DownloadFile(SizeOptions options) : base(options)
+        {
+            _fileUploader = new UploadFile(options);
```

Comment:
Should we instead extract the common code in some utility class and avoid the test interdependency?

Reply:
There isn't much common code. I am using UploadFile class to have all responsibility of uploading a file, which is a setup step for the download file test.
Maybe I can move the upload file functionality itself to a helper class and both tests can use that.

Reply:
I'd agree; it would be nice if the scenarios were as self-contained as possible.   You can probably inline the "upload a random stream" more easily than factoring the "upload a file" out and reusing.

Reply:
Sure, will do.